### ISSUE TITLE
Objects in rotated images become smaller than their corresponding boxes.

### ISSUE THREAD

**Issue by gH2AX** [ultralytics/ultralytics#16401]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I set the degrees parameter in my default.yaml file to 180 (setting the hyperparams in CLI did not work). The sample images in train_batch0jpg through train_batch2.jpg appear rotated but smaller than before; the corresponding bounding boxes stay the same size.

### Environment

Ultralytics YOLOv8.2.98 🚀 Python-3.11.10 torch-2.4.1+cu121 CUDA:0 (NVIDIA A100-PCIE-40GB, 40339MiB)
Setup complete ✅ (64 CPUs, 503.3 GB RAM, 7.6/40.0 GB disk)

OS                  Linux-4.18.0-372.16.1.el8_6.x86_64-x86_64-with-glibc2.28
Environment         Linux
Python              3.11.10
Install             pip
RAM                 503.35 GB
CPU                 Intel Core Processor (Haswell, IBRS)
CUDA                12.1

numpy               ✅ 1.26.4<2.0.0,>=1.23.0
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.4.1>=1.8.0
torchvision         ✅ 0.19.1>=0.9.0
tqdm                ✅ 4.66.5>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.6>=2.0.0
torch               ✅ 2.4.1!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

yolo detect train model=yolov10s.pt data=data.yaml \
    batch=7 epochs=83 imgsz=2202 project=results exist_ok=True \
    device=0 workers=2 > train.out

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @gH2AX, thank you for reaching out and sharing your experience with Ultralytics 🚀! This is an automated response to assist you while an Ultralytics engineer reviews your issue.

We see you're experiencing an issue with object sizes after rotation during training. If this is a 🐛 bug report, we kindly ask for a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to better diagnose the problem.

In the meantime, please consider:
- Checking if your YAML and other configurations align with the [Docs](https://docs.ultralytics.com).
- Ensuring your environment meets the [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), as sometimes version discrepancies can lead to unexpected behavior.

For more interactive support, join our [Discord](https://ultralytics.com/discord) community 🎧, or for detailed discussions, feel free to explore [Discourse](https://community.ultralytics.com).

## Upgrade

Ensure you are using the latest version by running:

```bash
pip install -U ultralytics
```

## Verified Environments

You can also run YOLOv8 in various verified environments to eliminate local issues:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

The CI status above ensures that all tests are passing across various platforms.

Thank you for your patience, and we'll get back to you soon! 🚀

**Comment by Y-T-G** [comment]
Do you have any example images?

**Comment by gH2AX** [comment]
> Do you have any example images?

Certainly. Here's two different transformations of the same original image. The one on the right is how I labelled it. On the left, you can see that the bounding box somehow expanded, or the object shrunk - I can't tell if it's an optical illusion.

![yolo_rotation_bug](https://github.com/user-attachments/assets/769f4e14-aeca-492c-bf7a-32154755021f)

Btw: Don't nag me about alternatives to achieve the same localisation task - I tried everything, including hough transforms, but not all images in the dataset share the same nice edges.

**Comment by Y-T-G** [comment]
It's non-reproducible because you didn't provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/).

**Comment by gH2AX** [comment]
> It's non-reproducible because you didn't provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/).

Ah, my bad. I'll see if I can strip it down to only the augmentation part and use publicly available images for that.

**Comment by glenn-jocher** [comment]
Thank you for understanding. Once you have a simplified example, feel free to share it, and we'll be happy to assist further.

**Comment by fengecho** [comment]
> > Do you have any example images?
> 
> Certainly. Here's two different transformations of the same original image. The one on the right is how I labelled it. On the left, you can see that the bounding box somehow expanded, or the object shrunk - I can't tell if it's an optical illusion.
> 
> ![yolo_rotation_bug](https://private-user-images.githubusercontent.com/92214854/369604320-769f4e14-aeca-492c-bf7a-32154755021f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxNTc4OTcsIm5iZiI6MTczMzE1NzU5NywicGF0aCI6Ii85MjIxNDg1NC8zNjk2MDQzMjAtNzY5ZjRlMTQtYWVjYS00OTJjLWJmN2EtMzIxNTQ3NTUwMjFmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAyVDE2Mzk1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3NjNlZGIyNTExODU2NzY4OThkMDk5NWFhNzI5OGE2NWFhYTI3NzNhNjhhZmVmMzU1MzI0MzUwZjVmNDRkNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uggCadty_jobP3uSAFYplZLzSj-eJtivI6XCljDXnjY)
> 
> Btw: Don't nag at me about alternatives to achieve the same localisation task - I tried everything, including hough transforms, but not all images in the dataset share the same nice edges.
> 
> Edit: Why am I getting the 'non-reproducible' label? I just reproduced it for the sake of sharing these images with you.

this is because the rotation augmentation made the original bbox become bigger，When you use rotation augmentation, the bounding box rotates by the same angle as the image. The new bounding box will enclose the old bounding box, which essentially makes the bounding box larger. You can try it yourself by drawing it out.
